paasta_tools/frameworks/task_store.py

```diff
@@ -0,0 +1,180 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import json
+
+from kazoo.client import KazooClient
+from kazoo.exceptions import NoNodeError
+
+from paasta_tools.utils import _log
+
+
+class MesosTaskParametersIsImmutableError(Exception):
+    pass
+
+
+class MesosTaskParameters(object):
+    def __init__(
+        self,
+        health=None,
+        mesos_task_state=None,
+        is_draining=None,
+        is_healthy=None,
+        offer=None,
+        resources=None,
+    ):
+        self.__dict__['health'] = health
+        self.__dict__['mesos_task_state'] = mesos_task_state
+        self.__dict__['is_draining'] = is_draining
+        self.__dict__['is_healthy'] = is_healthy
+        self.__dict__['offer'] = offer
+        self.__dict__['resources'] = resources
+
+    def __eq__(self, other):
+        return self.__dict__ == other.__dict__
+
+    def __repr__(self):
+        return "%s(\n    %s)" % (type(self).__name__, ',\n    '.join(["%s=%r" % kv for kv in self.__dict__.items()]))
+
+    def __setattr__(self, name, value):
+        raise MesosTaskParametersIsImmutableError()
+
+    def __delattr__(self, name):
+        raise MesosTaskParametersIsImmutableError()
+
+    def merge(self, **kwargs):
+        """Return a merged MesosTaskParameters object, where attributes in other take precedence over self."""
+
+        new_dict = self.__dict__
+        new_dict.update(kwargs)
+
+        return MesosTaskParameters(**new_dict)
+
+    @classmethod
+    def deserialize(cls, serialized_params):
+        return cls(**json.loads(serialized_params))
```

Comment:
nope, [json docs](https://docs.python.org/3/library/json.html) say it can be "a str, bytes or bytearray instance containing a JSON document"